006_buildkite_importing_pipelines/README.md

```diff
@@ -0,0 +1,374 @@
+<!-- TODO: curl isn't needed until the doc includes fetching the upstream schema
+- You have [`curl` installed](https://curl.se/dlwiz/), or can fetch a file from
+  a website some other way.
+-->
+
+## Steps
+
+### Convert YAML pipelines to CUE
+
+#### :arrow_right: Begin with a clean git state
+
+Change directory into the root of the repository that contains your Buildkite
+pipeline files, and ensure you start this process with a clean git state, with
+no modified files. For example:
+
+:computer: `terminal`
+```sh
+cd dependent-pipeline-example # our example repository
+git status                    # should report "working tree clean"
+```
+
+#### :arrow_right: Initialise a CUE module
+
+Initialise a CUE module named after the organisation and repository you're
+working with. For example:
+
+:computer: `terminal`
+```sh
+cue mod init github.com/buildkite/dependent-pipeline-example
+```
+
+#### :arrow_right: Import YAML pipelines
+
+Use `cue` to import your YAML pipeline files:
+
+:computer: `terminal`
+```sh
+cue import ./.buildkite/*.yml --with-context -p buildkite -f -l pipelines: -l 'strings.TrimSuffix(path.Base(filename),path.Ext(filename))'
+```
+
+Check that a CUE file has been created for each YAML pipeline in the
+`.buildkite` directory. For example:
+
+:computer: `terminal`
+```sh
+ls .buildkite/
+```
+
+Your output should look similar to this, with matching pairs of YAML and CUE
+files:
+
+```text
+pipeline.cue pipeline.deploy.cue pipeline.deploy.yml pipeline.yml
+```
+
+Observe that each pipeline has been imported into the `pipelines` struct, at a
+location derived from its original file name:
+
+:computer: `terminal`
+```sh
+head .buildkite/*.cue
+```
+
+The output should reflect your pipelines. In our example:
+
+```text
+==> .buildkite/pipeline.cue <==
+package buildkite
+
+pipelines: pipeline: steps: [{
+	command: "echo 'Tests'"
+	label:   ":hammer:"
+},
+	"wait", {
+		trigger:  "dependent-pipeline-example-deploy"
+		label:    ":rocket:"
+		branches: "master"
+
+==> .buildkite/pipeline.deploy.cue <==
+package buildkite
+
+pipelines: "pipeline.deploy": steps: [{
+	command:           "echo 'Deploy'"
+	label:             ":rocket:"
+	concurrency_group: "$BUILDKITE_PIPELINE_SLUG-deploy"
+	concurrency:       1
+	branches:          "master"
+}]
+```
+
+#### :arrow_right: Store CUE pipelines in a dedicated directory
+
+Create a directory called `buildkite` to hold your CUE-based Buildkite pipeline
+files. For example:
+
+:computer: `terminal`
+```sh
+mkdir -p internal/ci/buildkite
+```
+
+You may change the hierarchy and naming of `buildkite`'s **parent** directories
+to suit your repository layout. If you do so, you will need to adapt some
+commands and CUE code as you follow this guide.
+
+Move the newly-created CUE files into their dedicated directory. For example:
+
+:computer: `terminal`
+```sh
+mv ./.buildkite/*.cue internal/ci/buildkite
+```
+
+### Validate workflows
+
+<!-- TODO: upstream JSONSchema isn't usable
```

Comment:
See comment in the other PR. There is value to my mind to importing this schema today, because it imports cleanly if with somewhat looser constraints than we would like. That situation will improve however.